keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt

```diff
@@ -97,10 +97,17 @@ class EnvironmentPromotionChecker(
                        * want to request judgement or deploy a canary for artifacts that aren't
                        * deployed to a required environment or outside of an allowed time.
                        */
-                      val passesConstraints =
-                        checkStatelessConstraints(artifact, deliveryConfig, v, environment) &&
-                          checkStatefulConstraints(artifact, deliveryConfig, v, environment)
 
+                      val passesStatelessConstraints: Boolean = checkStatelessConstraints(artifact, deliveryConfig, v, environment)
+                      val passesStatefulConstraints: Boolean = checkStatefulConstraints(artifact, deliveryConfig, v, environment)
+
+                      log.info("for version: [$v] of artifact ${artifact.name}: " +
```

Comment:
added some logging statements, if it's not helpful I can remove it.

Reply:
I think they're definitely helpful! We typically don't have enough logs! But I'd lower the severity to `DEBUG`.